### PR TITLE
Update/paypal update locks

### PR DESCRIPTION
**Pull request by mikejolley**
Payment complete implementation plus some fixes noted when testing.

### CODE REVIEW

File: includes/class-wc-order.php [hunk 1]
```diff
@@ -103,25 +108,32 @@ public function payment_complete( $transaction_id = '' ) {
...
 				return false;
 			}
-			do_action( 'woocommerce_pre_payment_complete', $this->get_id() );
 
-			if ( ! empty( WC()->session ) ) {
-				WC()->session->set( 'order_awaiting_payment', false );
-			}
+			$lock_name = 'order_' . $this->get_id() . '_payment_complete';
+
+			if ( WC_Database_Locks::aquire_lock( $lock_name ) ) {
```
Comment (claudiulodro) [inline includes/class-wc-order.php]: If acquiring the lock fails, this function will return still `true`.

Comment (mikejolley) [inline includes/class-wc-order.php]: @claudiulodro No, it returns false. Only true once a lock is created successfully: https://github.com/woocommerce/woocommerce/pull/15967/files#diff-e63c6a8e54cbd89064a6f330abf1248cR107

Comment (claudiulodro) [inline includes/class-wc-order.php]: @mikejolley sorry I meant the `payment_complete` method that we're talking in right now. If acquiring the lock fails, `payment_complete` returns true.

**Comment by claudiulodro** [comment]
Looks good to me.

**Pull request merged**